src/core.c/CompUnit/Repository/Distribution.pm6

```diff
@@ -54,6 +57,24 @@ class CompUnit::Repository::Distribution does Distribution {
         }
     }
 
+    # Try to locate a distribution by a given file name. Only makes sense for CURFS.
+    # $file is expected to be either absolute or relative to $*CWD.
+    method from-file(::?CLASS:U: $file, :$name, :$ver, :$auth, :$api --> ::?CLASS:D) {
+        my @fs-repos = $*REPO.repo-chain.grep(CompUnit::Repository::FileSystem);
```

Comment:
If you have to grep for the type of repository then its not generalized enough

Reply:
Agree. I had too many other new concepts to grasp while doing all this, so this one just slipped out of my attention.

Do you have a good idea? I mean, I clearly didn't want the installations to be searched. The best alternative I see is `CompUnit::Repository::Locally && ! CompUnit::Repository::Installable`.

Reply:
At the very least it should be querying (despite the name `from-file` this is indeed effectively querying) a `CompUnit::Repository` object, not a distribution. Furthermore, since it is only ever called with the assumption that its for a precomp file, I would imagine the method (wherever it is) would be more appropriately be called something like `method distribution-from-precomp-file is implementation-detail { ... }` and potentially calling it via `.?distribution-from-precomp`. At least that is my initial thoughts.

Reply:
> since it is only ever called with the assumption that its for a precomp file

The assumption is wrong. The method is to be used for a file, not for a precomp. I.e. its purpose is to locate the distribution to which the file belongs.

Reply:
Addressed with fc4ee6e73309cefbe5784a48584a18e1f1024ce5